tracer/src/Datadog.Trace.SourceGenerators/Helpers/StringArrayComparer.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="StringArrayComparer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Datadog.Trace.SourceGenerators.Helpers;
+
+namespace Helpers
```

Comment:
NIT: Use scoped namespaces

Reply:
If this is going to be a requirement, we should add it to the style rules.

Reply:
We would like to using it going forwards, but we don't want it to fail on older files.